src/MetaTable.tsx

```diff
@@ -1,21 +1,21 @@
-import React, {useCallback, useEffect, useState} from 'react';
+import React, {useCallback, useMemo} from 'react';
 import {mergeData, renameColumns} from "./table-data-transform";
-import {DataDeep, Datum} from "gosling.js/dist/src/core/gosling.schema";
-import {GoslingRef} from "gosling.js";
-
+import type { Datum, DataDeep } from 'gosling.js/dist/src/gosling-schema';
 export type MetaTableSpec = {
     type: "table",
     // TODO: allow custom data specification for metatable
     data?: DataDeep;
     dataTransform: tableDataTransform[];
-    genomicColumns: [string,string] | [string];
+    genomicColumns: [string, string] | [string];
     columns?: string[];
```

Comment:
It might be good to clarify that this is for non-genomic columns.

```suggestion
    nonGenomicColumns?: string[];
```

Reply:
Or, you could merge this but have `type`, e.g., 
```ts
columns?: {  type: 'genomic' | 'nominal' | 'quantitative', columnName: string  }[];
```
This way, you could add additional properties for columns, e.g., formating of numbers

columns: [{ type: 'quantitative', columnName: 'score', format: '.2f' }] // d3.format()

Reply:
I decided to have genomics columns and metadata columns:
```typescript    
    genomicColumns: [string, string?];
    metadataColumns: { type: 'genomic' | 'nominal' | 'quantitative', columnName: string, columnFormat: string }[];
```
The genomic columns are explicit because they contain the start and (optionally) end of genomic features and at least the start is required.

I also kept the genomic option in the metadata columns. Maybe there is additional positional data that users want to see, e.g. if they visualize metadata for genes an additional genomic metadata column could be the position of the promotor.